encrypted-media/scripts/setmediakeys-multiple-times-with-different-mediakeys.js

```diff
@@ -53,34 +53,35 @@ function runTest(config, qualifier) {
             // Set mediaKeys2 on video element (switching MediaKeys) need not
             // fail after src attribute is set.
             return _video.setMediaKeys(_mediaKeys2);
-        })).then(test.step_func(function() {
+        }).then(test.step_func(function() {
             // Switching setMediaKeys after setting src attribute on video element
             // is not required to fail.
-            assert_equals(_video2.mediaKeys, _mediaKeys2);
             fail = false;
+            assert_equals(_video.mediaKeys, _mediaKeys2);
             return Promise.resolve();
-        }, test.step_func(function(error) {
+        }), test.step_func(function(error) {
             fail = true;
             assert_equals(_video.mediaKeys, _mediaKeys1);
             assert_in_array(error.name, ['InvalidStateError','NotSupportedError']);
             assert_not_equals(error.message, '');
             // Return something so the promise resolves properly.
             return Promise.resolve();
         })).then(function() {
-            // Set null mediaKeys on video (clearing MediaKeys) not
-            // supported after src attribute is set.
+            // Set null mediaKeys on video (clearing MediaKeys) is not required to fail
+            // after src attribute is set.
             return _video.setMediaKeys(null);
         }).then(test.step_func(function() {
-            assert_unreached('Clearing mediaKeys after setting src should have failed.');
+            assert_equals(_video.mediaKeys, null);
+            test.done();
         }), test.step_func(function(error) {
             if(fail) {
```

Comment:
`fail` is very specific. I think it should have a more specific name reflecting what exactly failed (or passed). Otherwise, it's not obvious why this `.then()` is the way it is.